File: java/src/main/java/ServletRequest.java
import java.util.Enumeration;

public interface ServletRequest {
    public static final String BASIC_AUTH = "BASIC";
    public static final String CLIENT_CERT_AUTH = "CLIENT_CERT";
    public static final String DIGEST_AUTH = "DIGEST";
    public static final String FORM_AUTH = "FORM";

    public String getAuthType();

    public String getContextPath();

    public String[] getParameterValues(String name);

    public long getDateHeader(String name);

    public String getHeader(String name);

    public Enumeration getHeaderNames();

    public Enumeration getHeaders(String name);

    public int getIntHeader(String name);

    public String getMethod();

    public String getPathInfo();

    public String getPathTranslated();

    public String getQueryString();

    public String getRemoteUser();

    public String getRequestedSessionId();

    public String getRequestURI();

    public StringBuffer getRequestURL();

    public String getServletPath();

    public boolean isRequestedSessionIdFromCookie();

    public boolean isRequestedSessionIdFromURL();

    public boolean isRequestedSessionIdValid();

    public boolean isUserInRole(String role);

    public boolean isRequestedSessionIdFromUrl();
}

File: php/src/ServletRequest.php
<?php

interface ServletRequest
{
    public const BASIC_AUTH = "BASIC";
    public const CLIENT_CERT_AUTH = "CLIENT_CERT";
    public const DIGEST_AUTH = "DIGEST";
    public const FORM_AUTH = "FORM";

    public function getAuthType(): string;

    public function getContextPath(): string;

    public function getParameterValues(String $name): array;

    public function getDateHeader(String $name);

    public function getHeader(String $name): string;

    public function getHeaderNames(): array;

    public function getHeaders(String $name): array;

    public function getIntHeader(String $name): int;

    public function getMethod(): string;

    public function getPathInfo(): string;

    public function getPathTranslated(): string;

    public function getQueryString(): string;

    public function getRemoteUser(): string;

    public function getRequestedSessionId(): string;

    public function getRequestURI(): string;

    public function getRequestURL(): string;

    public function getServletPath(): string;

    public function isRequestedSessionIdFromCookie(): bool;

    public function isRequestedSessionIdFromURL(): bool;

    public function isRequestedSessionIdValid(): bool;

    public function isUserInRole(String $role): bool;
}
